File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        
        HashMap <Integer, Integer>  map = new HashMap <>();
        
        for(int  t : tangerine) {
            if (!map.containsKey(t))  {
                map.put(t, 1);
            }else {
                map.put(t, map.get(t)+1);
            }
        }
        
        ArrayList <Integer> arr_li = new ArrayList <>(map.values());
        
        arr_li.sort (Collections.reverseOrder());
        int cur_cnt  = 0;
        int type_cnt = 0;
        
        for (int cnt : arr_li) {
            if (cur_cnt  >= k) {
                break;
            } 
            
            cur_cnt += cnt ;
            type_cnt +=1; 
        
        }
        answer = type_cnt;
        
        return answer;
        
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import Counter 

def solution(k, tangerine):
    
    answer = 0
    counter = Counter(tangerine)
    
    counts = sorted(counter.values(), reverse= True)
    # 귤 종류 최소 개수여야함.
    
    type_cnt = 0
    weight_sum = 0
    for cnt in counts:
        if weight_sum >= k :
            break
        weight_sum += cnt
        type_cnt +=1 
        
    answer= type_cnt
        
    
    
    
    return answer